test/amp-caching/amp-caching-test.js

```diff
@@ -0,0 +1,169 @@
+import { buildSW } from '../../index';
+import { promisify } from 'util';
+import { writeFile, unlink } from 'fs';
+import { join } from 'path';
+
+const promisifiedWriteFile = promisify(writeFile);
+const promisifiedUnlink = promisify(unlink);
+
+describe('AMP Caching Module', function() {
+  const driver = global.__AMPSW.driver;
+  const serviceWorkerPath = join('test', 'amp-caching-sw.js');
+  this.timeout(5000);
+
+  before(async () => {
+    const generatedSW = await buildSW();
+    await promisifiedWriteFile(serviceWorkerPath, generatedSW);
+    await driver.get('http://localhost:6881/test/index.html');
+  });
+
+  after(async () => {
+    await promisifiedUnlink(serviceWorkerPath);
+  });
+
+  beforeEach(async () => {
+    await driver.navigate().refresh();
+    await driver.executeAsyncScript(async cb => {
+      await window.__testCleanup();
+      const registration = await navigator.serviceWorker.register(
+        '/test/amp-caching-sw.js',
+      );
+      await window.__waitForSWState(registration, 'activated');
+      cb();
+    });
+    await driver.navigate().refresh();
+    await new Promise(resolve => setTimeout(resolve, 100));
```

Comment:
Both of these lines need a bit of explanation.

Is `100`ms just a lucky value? Can we instead listen to `document load` or similar on the page after refresh?

Reply:
this was just a timeout thing, removed this